[Wisp] Add KillThreads at RCMUnsafe and fix some errors and bugs.

ZhaiMo15 opened the pull request:
This patch contains 5 dragonwell11 patches.

Original patch url:
https://github.com/dragonwell-project/dragonwell11/commit/02606826e4fb63e2266e38c31c7d52c25d9a894c
https://github.com/dragonwell-project/dragonwell11/commit/ead121f9a070aba9863087fda5ca6d1e144df235
https://github.com/dragonwell-project/dragonwell11/commit/e2fbc0674908db5c440f2d5671539b8a4e4b7d15
https://github.com/dragonwell-project/dragonwell11/commit/105a3455f3c32c0c89293fd62139af84c05ce77f
https://github.com/dragonwell-project/dragonwell11/commit/69d7997da583f2c478dc7029b80ca1b73d55fcb2

Test Plan: all wisp tests

Reviewed-by: yulei

Issue:
https://github.com/dragonwell-project/dragonwell17/issues/138

---

[Wisp] Add KillThreads at RCMUnsafe
Summary: Add KillThreads function in RCMUnsafe to kill all
running threads inherited from the specified container.

Test Plan: jtreg TestKillThreads

---

[Wisp] AArch64: Fix TestUnsupportedArguments.sh for Wisp
Summary: Wisp has been supported on AArch64 platform so that we'd remove wisp arguments in this test

Test Plan: all wisp tests on x64 and aarch64 platform

---

[Wisp] Fix deoptimization when monitorexit calls java in synchronized methods or blocks

Summary: It has the possibility to get a deoptimization whenever entering into java code. 1. When this happens at calling Wisp.unpark at the bytecode _return of synchronized methods, it is going to crash or error because the origin deopt logic cannot handle the _return bytecode; 2. For synchronized blocks, the unlocked monitor shouldn't be recorded into the debuginfo so that an illegalMonitorStateException will be thrown out.

Test Plan: test/jdk/com/alibaba/wisp/monitor/deopt/*, which can reproduce this issue.

---

[Wisp] Fix work-stealing of JFR Event
Summary: The thread saved by `JfrConditionalFlushWithStacktrace` event should also be fixed.

Test Plan: TestJNICritical.java

---

[Misc] fix build error in debug mode
Summary: Fix build error in debug mode

Test Plan: CI tests

---

ZhaiMo15 committed:
[Wisp] Add KillThreads at RCMUnsafe.

Summary: Add KillThreads function in RCMUnsafe to kill all
running threads inherited from the specified container.

Test Plan: jtreg TestKillThreads

Reviewed-by: yulei

Issue:
https://github.com/dragonwell-project/dragonwell17/issues/138

ZhaiMo15 committed:
[Wisp] AArch64: Fix TestUnsupportedArguments.sh for Wisp.

Summary: Wisp has been supported on AArch64 platform so that we'd remove wisp arguments in this test

Test Plan: all wisp tests on x64 and aarch64 platform

Reviewed-by: yulei

Issue:
https://github.com/dragonwell-project/dragonwell17/issues/138

ZhaiMo15 committed:
[Wisp] Fix deoptimization when monitorexit calls java in synchronized methods or blocks.

Summary: It has the possibility to get a deoptimization whenever entering into java code. 1. When this happens at calling Wisp.unpark at the bytecode _return of synchronized methods, it is going to crash or error because the origin deopt logic cannot handle the _return bytecode; 2. For synchronized blocks, the unlocked monitor shouldn't be recorded into the debuginfo so that an illegalMonitorStateException will be thrown out.

Test Plan: test/jdk/com/alibaba/wisp/monitor/deopt/*, which can reproduce this issue.

Reviewed-by: yulei

Issue:
https://github.com/dragonwell-project/dragonwell17/issues/138

nebula-xm committed:
[Misc] Fix bugs in previous commit.

Summary:
1. Fix coroutine list dead lock.
2. Unify coroutine support lock.
3. Refine CoroutineSupportLocker usage locker name.
4. Fix coroutine support locker core in slowdebug mode.
5. Fix coroutine support locker core in coroutine dtor.

Test Plan:
all wisp cases

Reviewed-by: yulei

Issue:
https://github.com/dragonwell-project/dragonwell17/issues/138

ZhaiMo15 committed:
[Wisp] Fix work-stealing of JFR Event.

Summary: The thread saved by `JfrConditionalFlushWithStacktrace` event should also be fixed.

Test Plan: TestJNICritical.java

Reviewed-by: yulei

Issue:
https://github.com/dragonwell-project/dragonwell17/issues/138

yuleil commented on src/hotspot/share/runtime/coroutine.hpp:
```diff
@@ -491,6 +491,40 @@ class EnableStealMark: public StackObj {
 
+// Spin lock for cocurrent coroutine support.
+// It is useless at safepoint.
+class CoroutineSupportLocker : public StackObj {
```
Why do we need a spin lock? Isn't `Monitor` suitable?

nebula-xm replied:
`_coroutine_list` may be traversed not in safepoint. If this list changed by steal at this time, that will lead to deadlock. So we need a spin lock to resolve this problem. This locker use `cas` in previous version. So we create a spin lock to follow it.

ZhaiMo15 committed:
[Misc] Fix build error in debug mode.

Summary: Fix build error in debug mode

Test Plan: CI tests

Reviewed-by: yulei

Issue:
https://github.com/dragonwell-project/dragonwell17/issues/138
